### ISSUE TITLE
Button color in an active state should still reflect button type

### ISSUE THREAD

**Issue by kaitmore** [DecipherNow/gm-ui-components#154]
The green active state for the Button component overrides all button types (danger, warning, etc.) I would expect that a "danger" button in an active state would be some other shade of red.

**Comment by shanberg** [comment]
I realized we can generate the interaction style changes in the browser via `filter`. In testing, this bring the `generateButtonStyle` file down to about 50 lines, and we don't have to account for each style in it anymore.

```import { COLOR_WHITE } from "style/styleVariables";
import { contrastColor } from "style/styleFunctions";

/**
 * Utility function that transforms an object containing different attributes
 * into a block of CSS attributes dealing with buton styles
 *
 * @param {any} {
 *   buttonBackgroundColorBase = COLOR_WHITE,
 *   buttonBorderColorBase = COLOR_WHITE,
 *   buttonLabelColorBase = contrastColor(COLOR_WHITE, 100),
 * }
 * @returns string
 */
function generateButtonStyle({
  buttonBackgroundColorBase = contrastColor(COLOR_WHITE, 0),
  buttonBorderColorBase = contrastColor(COLOR_WHITE, 0.2),
  buttonLabelColorBase = contrastColor(COLOR_WHITE, 1),
}) {

  return `
  background-color: ${buttonBackgroundColorBase};
  border-color: ${buttonBorderColorBase};
  color: ${buttonLabelColorBase};
  filter: saturate(100%) brightness(100%);
  
  &:hover{
    filter: saturate(120%) brightness(95%);
  }

  &:active {
    transition-duration: 0s;
    filter: saturate(80%) brightness(90%) contrast(110%);
  }

  &:focus {
    z-index: 1;
  }

  &[disabled],
  &[disabled]:hover,
  &[disabled]:focus,
  &[disabled]:active {
    cursor: default;
    filter: saturation(0%);
  }
`;
}

export default generateButtonStyle;
```

**Comment by kaitmore** [comment]
Closed by #182
